src/main/c/cext/ruby.c

```diff
@@ -3235,8 +3228,16 @@ VALUE rb_enc_reg_new(const char *s, long len, rb_encoding *enc, int options) {
   rb_tr_error("rb_enc_reg_new not implemented");
 }
 
-VALUE rb_enc_vsprintf(rb_encoding *enc, const char *fmt, va_list ap) {
-  rb_tr_error("rb_enc_vsprintf not implemented");
+VALUE rb_enc_vsprintf(rb_encoding *enc, const char *format, va_list args) {
+  // TODO CS 7-May-17 this needs to use the Ruby sprintf, not C's
+  char *buffer;
+  if (vasprintf(&buffer, format, args) < 0) {
+    rb_tr_error("vasprintf error");
+  }
+  VALUE string = rb_str_new_cstr(buffer);
+  rb_enc_associate(string, enc);
```

Comment:
These two lines are basically what `rb_enc_str_new_cstr()` do I think, but it's not yet implemented.

Reply:
Interesting, thanks. FYI, you might've noticed, but I made the caller explicitly pass the `ASCII-8BIT` encoding. MRI passes NULL, which I think it then treats as "encoding index" 0, which happens to correspond to `ASCII-8BIT`. That might explain their call chain, that I semi-followed.